### DISCUSSION TITLE
Nextjs: SSR support - How we render Lit React Components in Nextjs with SSR?

### DISCUSSION THREAD

**Discussion by 2manoj1** [lit/lit#2245] [Q&A]
We need to render React web components which was build with Lit React. Now we need use those components in Nextjs app but we need render them as SSR. I tried below its only working client side rendering with Nextjs dynamic.  Any solutions for running Lit web components as SSR with Nextjs? 

```js 
import dynamic from "next/dynamic";
// not work ssr
// import { MButton } from "mm-react-web-components";

// clint side render
const MButton = dynamic(
  () => import("mm-react-web-components").then(({ MButton }) => MButton),
  {
    loading: () => <p>MButton Loading</p>,
    ssr: false,
  }
);
```
<img width="1130" alt="Screenshot 2021-10-13 at 2 56 46 PM" src="https://user-images.githubusercontent.com/88817537/137108785-fca4615f-2a18-4001-ae58-1e28054e23c8.png">

**Comment by justinfagnani** [comment]
I think you've done the right thing for now with `dynamic()`. Since Lit itself depends on DOM APIs, there's not too much we can do without Lit SSR support wired in. There may be a minimal support level we can get working where we load the element definition, but don't access the DOM APIs outside of the element instance, and get a fallback to React rendering elements and attributes.

**Comment by manandkumaar** [comment]
@justinfagnani Thanks for the info. 

We are build design system components using `lit-html` and `@lit-labs/react` which works perfectly in browser. 
But the same is not working in SSR with Next.JS.

I found from different issues discussions that Lit is not ready yet for SSR and new package `lit SSR` is getting built.
Is it possible to tell the road map of Lit SSR with next.js/remix?.

**Reply by aomarks** [reply]
We are actively working on Next integration. You can follow along at https://github.com/lit/lit/issues/2391.

**Reply by manandkumaar** [reply]
Thanks @aomarks for the update.

**Comment by manandkumaar** [comment]
Hi @e111077 and @justinfagnani ,

As discussed, moving our conversation from slack to this discussion. 
Looking forward to hear more on the SSR/SSG timeline and roadmap.
Thanks.

@aomarks @justinfagnani FYI. 
Elliott is on vacation. It is much appreciated if can able to share SSR/SSG roadmap and timeline.
